embassy-sync/src/pipe.rs

```diff
@@ -293,7 +293,19 @@ where
     pub fn write<'a>(&'a self, buf: &'a [u8]) -> WriteFuture<'a, M, N> {
         WriteFuture { pipe: self, buf }
     }
-
+    
+    /// Write all bytes to the pipe.
+    ///
+    /// This method writes all bytes from `buf` into the pipe
+    pub async fn write_all<'a>(&'a self, buf: &'a [u8]) -> Result<(), TryWriteError> {
```

Comment:
- explicitly writing the `'a` lifetime is not needed
- this can't fail, so it should return nothing, not a Result.